code/modules/mechanics/MechanicMadness.dm

```diff
@@ -502,11 +501,24 @@
 	proc/componentSay(var/string)
 		string = trim(sanitize(html_encode(string)))
 		var/maptext = null
-		var/mob/user = usr
-		if (src_exists_inside_user_or_user_storage && !src.storage)
-			maptext = make_chat_maptext(src.owner, "[string]", "color: #FFBF00;", alpha = 255)
+		var/loc = null //Location used for center of all_hearers scan "Probably where you want your text attached to."
+
+		if(src.stored?.linked_item && !src.storage) //If the sound synth is attached inside something.
+			if(istype_exact(src.stored?.linked_item, /obj/item/storage/mechanics/housing_large)) //Mech Cabinet check
+				loc = src.loc
+
+			else if(istype_exact(src.stored?.linked_item, /obj/item/storage/mechanics/housing_handheld)) //Device Frame check
+				if(!obj_loc_chain(src)) //If not stored anywhere usually in your hands or on the ground
+					loc = src.loc
```

Comment:
This condition will never be true because obj_loc_chain returns a list with is "truthy" in byond even if empty.